### DISCUSSION TITLE
Ignore a particular line (`# pyright: ignore`)

### DISCUSSION THREAD

**Discussion by savchenko** [microsoft/pyright#8325] [Q&A] [answered]
I am using pyright with NeoVim via `lspconfig`. In code, I have the following line:
```py
from . import routes  # pyright: ignore
```
Despite the ignore instruction, pyright LSP sets the following:
```
DiagnosticUnnecessary vim.lsp.pyright.2/diagnostic/underline
```
Similarly, `# noqa` does nothing (which is somewhat expected given the https://github.com/microsoft/pyright/issues/144).

How do I tell pyright to ignore all imports, but _only for a particular line_?

**Comment by erictraut** [comment]
There is no way to individually suppress tagged hints. These are not intended to be displayed as regular diagnostics. They are unaffected by `# pyright: ignore`, which affect only regular diagnostics. Clients should render tagged hints as "dimmed" text. See VS Code for the intended behavior. Some clients unfortunately render tagged hints as regular diagnostics. If your client does this, you can tell pyright to disable all tagged hints by setting `pyright.disableTaggedHints` to true. See [this documentation](https://microsoft.github.io/pyright/#/settings) for a full list of language server settings.

**Reply by savchenko** [reply]
Here is how it would render it:
![image](https://github.com/microsoft/pyright/assets/300146/1f08d85b-8e3b-4a73-ad3f-5de8c69d74f0)

Of course, vim being vim there is plethora of possible options, including the in-line display:
![image](https://github.com/microsoft/pyright/assets/300146/af3f8dc8-9022-4d5a-b264-2d9b3f39c58f)

I can disable the text altogether, but this doesn't prevent the hint (`H` on the side, designated as `DiagnosticSignHint`) and highlight:
![image](https://github.com/microsoft/pyright/assets/300146/6e9c37a2-37a8-4256-80cf-bd7d54be4348)

It can be disabled globally of course, but again, this is not my intention.

Any particular reason why they can't (shouldn't?) be individually "toggleable"?

**Reply by erictraut** [reply, marked as answer]
> Any particular reason why they can't (shouldn't?) be individually "toggleable"?

Because these are not intended to be diagnostics. They're meant to be rendered as a subtle visual clue that a symbol is unreferenced. If rendered correctly, you should not feel compelled to "disable" or suppress these.

Here's how it renders in VS Code for comparison. There is no "underline squiggle" or other indication that there is a diagnostic presented on that line because a "tagged hint" isn't meant to be a real diagnostic.
![image](https://github.com/microsoft/pyright/assets/7040122/ea352a56-4142-4cd1-a848-74450d9172d5)

For a full history of this discussion, refer to [this thread](https://github.com/microsoft/pyright/issues/1118).
